apis/telemetry/v1alpha1/metricpipeline_conversion.go

```diff
@@ -3,233 +3,48 @@ package v1alpha1
 import (
 	"errors"
 
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 
 	telemetryv1beta1 "github.com/kyma-project/telemetry-manager/apis/telemetry/v1beta1"
 )
 
-// Converts between v1alpha1 and v1beta1 MetricPipeline CRDs
-// There are no major changes to the MetricPipeline API between v1alpha1 and v1beta1.
-// However, some changes were done in shared types which are documented in the related file and require to convert MetricPipelines.
-
 var errSrcTypeUnsupportedMetricPipeline = errors.New("source type is not MetricPipeline v1alpha1")
 var errDstTypeUnsupportedMetricPipeline = errors.New("destination type is not MetricPipeline v1beta1")
 
-// ConvertTo implements conversion.Hub for MetricPipeline (v1alpha1 -> v1beta1)
-func (src *MetricPipeline) ConvertTo(dstRaw conversion.Hub) error {
-	dst, ok := dstRaw.(*telemetryv1beta1.MetricPipeline)
-	if !ok {
-		return errDstTypeUnsupportedMetricPipeline
-	}
-
-	// Copy metadata
-	dst.ObjectMeta = src.ObjectMeta
-
-	// Copy input fields
-	dst.Spec.Input = telemetryv1beta1.MetricPipelineInput{}
-	if src.Spec.Input.Prometheus != nil {
-		dst.Spec.Input.Prometheus = &telemetryv1beta1.MetricPipelinePrometheusInput{
-			Enabled:           src.Spec.Input.Prometheus.Enabled,
-			Namespaces:        convertNamespaceSelectorToBeta(src.Spec.Input.Prometheus.Namespaces),
-			DiagnosticMetrics: convertDiagnosticMetricsToBeta(src.Spec.Input.Prometheus.DiagnosticMetrics),
-		}
+func Convert_v1alpha1_MetricPipelineRuntimeInput_To_v1beta1_MetricPipelineRuntimeInput(in *MetricPipelineRuntimeInput, out *telemetryv1beta1.MetricPipelineRuntimeInput, s apiconversion.Scope) error {
```

Comment:
nitpick(not necessary): it would be nice to move this function after the `ConvertTo` and `ConvertFrom` functions to have same ordering as in the logpipeline_conversion.go